linalg/operator.hpp

```diff
@@ -433,6 +433,81 @@ class ConstrainedOperator : public Operator
    virtual ~ConstrainedOperator() { if (own_A) { delete A; } }
 };
 
+
+/** @brief Mimic the action of a complex operator using two real operators.
+
+    This operator requires vectors that are twice the length of its internally
+    stored real operators, Op_Real and Op_Imag. It is assumed that these vectors
+    store the real part of the vector first followed by its imaginary part.
+
+    ComplexOperator allows one to choose a convention upon construction, which
+    facilitates symmetry.
+
+    Matrix-vector products are then computed as:
+
+    1. When Convention::HERMITIAN is used (default)
+    / y_r \   / Op_r -Op_i \ / x_r \
+    |     | = |            | |     |
+    \ y_i /   \ Op_i  Op_r / \ x_i /
+
+    2. When Convention::BLOCK_SYMMETRIC is used
+    / y_r \   / Op_r -Op_i \ / x_r \
+    |     | = |            | |     |
+    \-y_i /   \-Op_i -Op_r / \ x_i /
```

Comment:
The minus in front of the output vector `y_i` is not correct. I think you mean `y_i = -Im(A*x) = -A_r*x_i - A_i*x_r` as opposed to `y_i = Im(A*x) = A_r*x_i + A_i*x_r` in the first case. Or, do you mean something else?

Reply:
It is confusing I agree, but I think the idea was to stress that with this convention one gets conjugated output unless one conjugates input vector explicitly.